tasks/tiler/xyz.py

```diff
@@ -34,24 +35,29 @@ def get_config_file(self):
     def get_geography_level(self, level):
         raise NotImplementedError('Geography levels file must be implemented by the child class')
 
-    def requires(self):
-        return {
-            'shorelineclip': ShorelineClip(geography=self.geography, year='2015')
-        }
-
     def run(self):
         config_data = self._get_config_data()
         for table_config in config_data:
             table_schema = self._get_table_schema(table_config)
             table_bboxes = table_config['bboxes']
             self._create_schema_and_table(table_schema)
-            self._generate_and_insert_tiles(self.zoom_level, table_schema, table_config['columns'], table_bboxes)
+            self._generate_and_insert_tiles(self.zoom_level, table_schema, table_config, table_bboxes)
+
+    def _get_table_name(self, table_schema):
+        return table_schema['table_name'] + '_' + self.month
+
+    def _get_mc_dates(self):
+        session = current_session()
+        query = '''
+                SELECT cdb_observatory.OBS_GetMCDates('{schema}', '{geo_level}', '{month}');
+                '''.format(schema='us.mastercard', geo_level=self.geography, month=self.month)
+        return session.execute(query).fetchone()[0]
```

Comment:
Here we're returning all the dates. Why don't we only use the month? O you want to leave this open in order to ask for a defined date for example 10/10/2017?

Reply:
I would like to be able to ask for a given date (as it's not very well defined right now I wanted it to be a bit more open, but we can revisit this later if needed)